__init__.py

```diff
@@ -268,28 +168,69 @@ def lookup(auto_suggest):
             res_auto_suggest, res_without_auto_suggest = (
                 list(pool.map(lookup, (True, False)))
             )
-        # Check the results, return PageMatch (autosuggest
-        # preferred) otherwise return the autosuggest
-        # PageDisambiguation.
-        if ((isinstance(res_auto_suggest, PageDisambiguation) or
-             (res_auto_suggest is None)) and
-                isinstance(res_without_auto_suggest, PageMatch)):
-            ret = res_without_auto_suggest
-        else:
-            ret = res_auto_suggest
+        ret = res_without_auto_suggest
         return ret
 
-    def display_article(self, match):
-        """Display the match page on a GUI if connected.
+    def respond(self, query):
+        result = self.get_wiki_result(query)
+        if result is not None:
+            if isinstance(result, PageMatch):
+                result = result.summary
+            elif isinstance(result, PageDisambiguation):
+                # we auto disambiguate here 
+                if len(result.options) > 0:
+                    result = self.get_wiki_result(result.options[0])
+                    if result is not None:
+                        result = result.summary
+                else:
+                    result = None
+
+        return result
+
+    def fix_input(self, query):
+        for noun in self.question_words:
+            for verb in self.question_verbs:
+                for article in [i + ' ' for i in self.articles] + ['']:
+                    test = noun + verb + ' ' + article
+                    if query[:len(test)] == test:
+                        return query[len(test):]
+        return query
+
+    def CQS_match_query_phrase(self, query):
+        answer = None
+        test = self.fix_input(query)
+
+        if test is not None:
+            answer = self.respond(test)
+
+        if answer:
+            return (query, CQSMatchLevel.CATEGORY, answer)
+        return answer
+
+    def stop(self):
+        pass
+
+    @intent_handler(IntentBuilder("AskWiki").require("Wikipedia"))
+    def handle_ask_wiki(self, message):
+        """entry point when wiki is called out by name
+           in the utterance"""
+        utt = message.data['utterance']
+
```

Comment:
It would be good to have some sort of feedback here like previously there was "I'm searching wikipedia for..." since it's a rather slow operation.

But again I assume there has been some discussion about the UI design around this so feel free to disregard this since it's merely an opinion

Reply:
I used to have it say, 'searching wikipedia for bla' but then I moved that output below before the answer. This seemed correct when wiki fails to return an answer in which case the total dialog is the unknown dialog. It seemed weird it would say searching 'wiki for bla', then say sorry <unknown random response>. Its a minor change and I have no opinion. I figured I'd get it in Derick's hands and let him tell me if he agrees or not.